subprojects/docs/src/docs/release/notes.md

```diff
@@ -24,6 +24,13 @@ For Java, Groovy, Kotlin and Android compatibility, see the [full compatibility
 
 <!-- Do not add breaking changes or deprecations here! Add them to the upgrade guide instead. --> 
 
+## Build reliability improvements
+
+Gradle employs a number of optimizations to ensure that builds are executed as fast as possible.
+These optimizations rely on the inputs and outputs of tasks to be well-defined.
+If a task is found to be invalid, Gradle will now execute it without the benefit of parallel execution, up-to-date checks and the build cache.
```

Comment:
```suggestion
Gradle already applies some validation to tasks to check whether they are well defined.
If a task is found to be invalid, Gradle will now execute it without the benefit of parallel execution, up-to-date checks and the build cache.
```

It is unclear to me what `invalid` means from that sentence. I suggest adding a sentence that Gradle already does some validation.